### PR TITLE
[deps] Upgrade terser-webpack-plugin to 5.x and fix compatibility issues #936

### LINKED ISSUE #936: [deps] Upgrade terser-webpack-plugin to 5.x and fix incompatibility issues
**Describe the bug**
Commit 838338bfd69186667bc4726b466db26b82aefae3 was reverted in #934  because `terser-webpack-plugin@5` does not support `TerserPlugin.uglifyJsMinify`. When running the application with this configuration, the Web Login Page (WLP) throws the error:
> Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate.

**Root cause**
The upgrade to `terser-webpack-plugin@5` and usage of `TerserPlugin.uglifyJsMinify` led to minification issues, resulting in runtime errors in React components.

**Workaround / Proposed Fix**
The following change to `webpack.config.js` resolves the error, but needs further validation to ensure there are no unintended side effects:

```diff
diff --git a/config/webpack.config.js b/config/webpack.config.js
index 7513b31..c3e2cc8 100644
--- a/config/webpack.config.js
+++ b/config/webpack.config.js
@@ -60,8 +60,7 @@ module.exports = (env, argv) => {
   if (argv.mode === "production") {
     cssLoaders = [MiniCssExtractPlugin.loader, "css-loader"];
     minimizers = [
-      new CssMinimizerPlugin(),
-      new TerserPlugin({minify: TerserPlugin.uglifyJsMinify}),
+      new CssMinimizerPlugin(), new TerserPlugin(),
     ];
     setup.removeDefaultConfig();
     plugins.push(
```

**Next steps**
- Verify the fix above does not introduce regressions or other issues, especially regarding JS minification and bundle correctness.
- Consider adding tests or steps to reproduce the `Maximum update depth exceeded` error for future reference.

### PR DESCRIPTION
**Pull request by Eeshu-Yadav**
## Checklist

- [x] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).
- [x] I have manually tested the changes proposed in this pull request.
- [ ] I have written new test cases for new code and/or updated existing tests for changes to existing code.
- [ ] I have updated the documentation.

Upgraded terser-webpack-plugin from 4.2.3 to 5.3.16 and fixed compatibility issues. Updated webpack config to use 'format' instead of deprecated 'output' option.

Fixes #936

### CODE REVIEW

**Review by nemesifier — COMMENTED** [review]
@Eeshu-Yadav the CI build fails

**Review by pandafy — APPROVED** [review]
LGTM!
